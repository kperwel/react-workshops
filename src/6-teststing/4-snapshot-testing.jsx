import React from "react";

export const Component = ({ name, description, onButtonClick }) => (
  <div>
    <h1>{name}</h1>
    <p>{description}</p>
    <button onClick={() => onButtonClick(name)}>User action button</button>
  </div>
);
