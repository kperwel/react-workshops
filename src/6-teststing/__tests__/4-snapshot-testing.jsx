import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { render, fireEvent } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import { Component } from "../4-snapshot-testing";

test("Should match provided snapshot", async () => {
  const { asFragment } = render(
        <Component />
  );

  expect(asFragment()).toMatchSnapshot()
});
