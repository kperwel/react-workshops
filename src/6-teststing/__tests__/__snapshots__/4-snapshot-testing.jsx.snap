// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match provided snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1 />
    <p />
    <button>
      User action button
    </button>
  </div>
</DocumentFragment>
`;
